packages/flutter_tools/lib/src/project.dart

```diff
@@ -975,12 +1004,13 @@ to migrate your project.
       logger: globals.logger,
       templateRenderer: globals.templateRenderer,
     );
+    final String androidIdentifier = parent.manifest.androidPackage ?? 'com.example.${parent.manifest.appName}';
```

Comment:
This seems like a good fallback, already used in https://github.com/flutter/flutter/blob/e7b7ebc066c1b2a5aa5c19f8961307427e0142a6/packages/flutter_tools/lib/src/android/deferred_components_prebuild_validator.dart#L271